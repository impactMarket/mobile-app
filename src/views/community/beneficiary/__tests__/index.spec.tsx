import { assert } from 'chai';
import { shallow, ShallowWrapper, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { clear } from 'jest-date-mock';
import * as React from 'react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

import BeneficiaryScreen from '../index';

const mockStore = configureMockStore();
const store = mockStore({});

configure({ adapter: new Adapter() });

describe('Beneficiary screen test', () => {
    let wrapper: ShallowWrapper;
    let navigation: any;

    beforeEach(() => {
        jest.resetAllMocks();
    });

    afterEach(() => {
        clear();
    });

    it('should render beneficiary screen correctly', () => {
        givenComponent();
        thenItRenderProperly();
    });

    it('should exists', function () {});

    function givenComponent() {
        assert.isDefined(BeneficiaryScreen);

        wrapper = shallow(
            <Provider store={store}>
                <BeneficiaryScreen />
            </Provider>
        );
    }

    function thenItRenderProperly() {
        expect(wrapper).toMatchSnapshot();
    }
});
