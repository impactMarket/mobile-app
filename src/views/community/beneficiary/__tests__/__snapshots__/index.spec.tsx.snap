// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Beneficiary screen test suite should render beneficiary screen correctly 1`] = `ShallowWrapper {}`;
